public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -98,8 +99,55 @@ export class PanelQueryRunner {
           let fieldConfig = this.dataConfigSource.getFieldOverrideOptions();
           let processFields = fieldConfig != null;
 
+          // TODO: create new "chunk" dataframe type ?
+          const firstFrame = data.series[0];
```

Comment:
I need to easily differentiate between a standard `DataFrame` containing all of the data and a "chunk" of a `DataFrame` which is supposed to be appended to the buffer on the main thread. I see multiple options:


1. adding a new type to `DataQueryResponseData` https://github.com/grafana/grafana/blob/7319efe0779d97c8dac5b0e6d9a3749b177c45a0/packages/grafana-data/src/types/datasource.ts#L428
2. setting up some convention on `DataQueryResponse.key` https://github.com/grafana/grafana/blob/7319efe0779d97c8dac5b0e6d9a3749b177c45a0/packages/grafana-data/src/types/datasource.ts#L437-L442 (e.g. `stream/12512521/full` for "full frames" and `stream/12512521/chunk` for "chunks"
3. adding new fields to `DataQueryResponse` https://github.com/grafana/grafana/blob/7319efe0779d97c8dac5b0e6d9a3749b177c45a0/packages/grafana-data/src/types/datasource.ts#L430 (e.g. `streamingChunk?: boolean`)
4. changing the shape of `DataQueryResonse` depending on `state`, e.g. 


```

export interface StreamingDataQueryResponse {
  data: (Chunk | DataFrameDto)[];
  key: string;
  error?: DataQueryError;
  state: LoadingState.Streaming;
}
```


I prefer 4th option as its the most explicit, thoughts? @ryantxu @grafana/grafana-edge-squad

Reply:
:+1: